src/controllers/setup_test.go

```diff
@@ -0,0 +1,59 @@
+package controllers_test
+
+import (
+	"context"
+	"testing"
+
+	cappv1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+	runtimeFake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var (
+	client    *fake.Clientset
+	dynClient runtimeClient.WithWatch
+	logger    *zap.Logger
+)
+
+const (
+	testNamespace = "test-namespace"
+	testName      = "test"
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	createTestNamespace(testNamespace)
```

Comment:
ill create the namespace inside each test function because you want the test to be isolated.

Reply:
Handling this in #80